packages/nimble-components/src/select/specs/filterable-select-hld.md

```diff
@@ -0,0 +1,115 @@
+# Filterable Select
+
+## Problem Statement
+
+Our clients have a need of a filterable dropdown component that does not allow arbitrary text as a value, but only values available in the dropdown. The filtering will have the following behaviors (in the initial feature set):
+
+-   The filter text will match _any_ text within the `textContent` of each `ListOption` in a case insensitive way. Specifically, the filter text will match wholly against the target (e.g. filter text of "ad" will match "Add", but not "abcd").
+-   Each time the dropdown is opened the filter text is cleared.
+-   While the dropdown is opened all keystrokes except `<ArrowUp>`. `<ArrowDown>`, `<Enter>`, `<Home>` and `<End>` will apply to the filter.
+-   Pressing `<Esc>` will close the dropdown and revert the value to what it was prior to opening dropdown.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [#1273: Filterable NimbleSelect](https://github.com/ni/nimble/issues/1273)
+-   [Filterable Select IxD](./IxD.md#filter)
+
+## Implementation / Design
+
+### API
+
+#### Select
+
+We will provide a means for clients to enable this feature with a new attribute:
+
+```ts
+export class Select() {
+    ...
+    @attr({ attribute: 'filter-mode' })
+    public filterMode = FilterMode.none;
+    ...
+}
+
+// Provide enum for filterMode to allow for future modes
+export const FilterMode  = {
+    none: 'none';
```

Comment:
nit: the value of the default enum case should be `undefined` and not a string. https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#attribute-common-value-patterns